lib/action_controller/serialization.rb

```diff
@@ -27,6 +27,7 @@ def get_serializer(resource, options = {})
       end
       ActiveModel::SerializableResource.serialize(resource, options) do |serializable_resource|
         if serializable_resource.serializer?
+          serializable_resource.url_helper = self
```

Comment:
You can do that, but what we are trying to is more like `Rails.application.routes.url_for([@post, :comments])`